src/details/ArborX_DetailsBruteForceImpl.hpp

```diff
@@ -53,26 +53,94 @@ struct BruteForceImpl
   static void query(ExecutionSpace const &space, Primitives const &primitives,
                     Predicates const &predicates, Callback const &callback)
   {
+    using TeamPolicy = Kokkos::TeamPolicy<ExecutionSpace>;
     using AccessPrimitives = AccessTraits<Primitives, PrimitivesTag>;
     using AccessPredicates = AccessTraits<Predicates, PredicatesTag>;
+    using PredicateType = typename AccessTraitsHelper<AccessPredicates>::type;
+    using PrimitiveType = typename AccessTraitsHelper<AccessPrimitives>::type;
 
     int const n_primitives = AccessPrimitives::size(primitives);
     int const n_predicates = AccessPredicates::size(predicates);
+    int max_scratch_size = TeamPolicy::scratch_size_max(0) / 2;
+    int const predicates_per_team = max_scratch_size / sizeof(PredicateType);
```

Comment:
May want to add a check that `predicates_per_team` is not zero. I think this could happen if we accidentally attach a very heavy data to a predicate by accident.

Reply:
This was done